Device Service panic when bootstrap failed

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1432ddd]

goroutine 1 [running]:
github.com/edgexfoundry/device-sdk-go/v2/pkg/service.(*DeviceService).Stop(0xc0003340f0, 0xc00032c300)
        /Users/chris/Workspace/device-sdk-go/pkg/service/service.go:136 +0x3d
github.com/edgexfoundry/device-sdk-go/v2/pkg/service.Main(0x151259e, 0xd, 0x1518ef7, 0x1a, 0x14d2280, 0xc000260f90, 0x15ac120, 0xc00032c300, 0xc000309200, 0xc000342000)
        /Users/chris/Workspace/device-sdk-go/pkg/service/main.go:73 +0x6fc
github.com/edgexfoundry/device-sdk-go/v2/pkg/startup.Bootstrap(0x151259e, 0xd, 0x1518ef7, 0x1a, 0x14d2280, 0xc000260f90)
        /Users/chris/Workspace/device-sdk-go/pkg/startup/bootstrap.go:19 +0x10e
main.main()
        /Users/chris/Workspace/device-sdk-go/example/cmd/device-simple/main.go:23 +0x74
```

this is because DeviceService.manager might not be initialized yet during bootstrap sequence.

https://github.com/edgexfoundry/device-sdk-go/blob/e2b795a37b7f4058c403051823a7e04e4e702841/pkg/service/service.go#L133

--- a/internal/autoevent/manager.go
+++ b/internal/autoevent/manager.go
@@ -63,18 +63,6 @@ func (m *manager) StartAutoEvents() {
 	}
 }
 
-func (m *manager) StopAutoEvents() {
-	m.mutex.Lock()
-	defer m.mutex.Unlock()
-
-	for deviceName, executors := range m.executorMap {
-		for _, executor := range executors {
-			executor.Stop()
-		}
-		delete(m.executorMap, deviceName)
-	}
-}
-
 func (m *manager) triggerExecutors(deviceName string, autoEvents []models.AutoEvent, dic *di.Container) []*Executor {
 	var executors []*Executor
 	lc := bootstrapContainer.LoggingClientFrom(dic.Get)

--- a/pkg/models/manager.go
+++ b/pkg/models/manager.go
@@ -8,8 +8,6 @@ package models
 type AutoEventManager interface {
 	// StartAutoEvents starts all the AutoEvents of the device service
 	StartAutoEvents()
-	// StopAutoEvents stops all the AutoEvents of the device service
-	StopAutoEvents()
 	// RestartForDevice restarts all the AutoEvents of the specific device
 	RestartForDevice(name string)
 	// StopForDevice stops all the AutoEvents of the specific device

--- a/pkg/service/init.go
+++ b/pkg/service/init.go
@@ -8,19 +8,15 @@ package service
 
 import (
 	"context"
-	"fmt"
-	"net/http"
 	"sync"
-	"time"
 
 	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/startup"
 	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
 	"github.com/gorilla/mux"
 
-	v2cache "github.com/edgexfoundry/device-sdk-go/v2/internal/cache"
-	"github.com/edgexfoundry/device-sdk-go/v2/internal/container"
+	"github.com/edgexfoundry/device-sdk-go/v2/internal/cache"
 	"github.com/edgexfoundry/device-sdk-go/v2/internal/provision"
-	dsModels "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	"github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
 )
 
 // Bootstrap contains references to dependencies required by the BootstrapHandler.
@@ -37,63 +33,49 @@ func NewBootstrap(router *mux.Router) *Bootstrap {
 
 func (b *Bootstrap) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, startupTimer startup.Timer, dic *di.Container) (success bool) {
 	ds.UpdateFromContainer(b.router, dic)
+	ds.controller.InitRestRoutes()
 
-	err := ds.selfRegister()
+	err := cache.InitV2Cache(ds.ServiceName, dic)
 	if err != nil {
-		ds.LoggingClient.Error(fmt.Sprintf("Couldn't register to metadata service: %v\n", err))
-		return false
-	}
-
-	err = v2cache.InitV2Cache(ds.ServiceName, dic)
-	if err != nil {
-		ds.LoggingClient.Errorf("failed to init cache: %v", err)
+		ds.LoggingClient.Errorf("Failed to init cache: %v", err)
 		return false
 	}
 
 	if ds.AsyncReadings() {
-		ds.asyncCh = make(chan *dsModels.AsyncValues, ds.config.Service.AsyncBufferSize)
+		ds.asyncCh = make(chan *models.AsyncValues, ds.config.Service.AsyncBufferSize)
 		go ds.processAsyncResults(ctx, wg, dic)
 	}
 	if ds.DeviceDiscovery() {
-		ds.deviceCh = make(chan []dsModels.DiscoveredDevice, 1)
+		ds.deviceCh = make(chan []models.DiscoveredDevice, 1)
 		go ds.processAsyncFilterAndAdd(ctx, wg)
 	}
 
-	err = ds.driver.Initialize(ds.LoggingClient, ds.asyncCh, ds.deviceCh)
-	if err != nil {
-		ds.LoggingClient.Error(fmt.Sprintf("Driver.Initialize failed: %v\n", err))
+	e := ds.driver.Initialize(ds.LoggingClient, ds.asyncCh, ds.deviceCh)
+	if e != nil {
+		ds.LoggingClient.Errorf("Failed to init ProtocolDriver: %v", e)
 		return false
 	}
 	ds.initialized = true
 
-	dic.Update(di.ServiceConstructorMap{
-		container.DeviceServiceName: func(get di.Get) interface{} {
-			return ds.deviceService
-		},
-		container.ProtocolDiscoveryName: func(get di.Get) interface{} {
-			return ds.discovery
-		},
-		container.ProtocolDriverName: func(get di.Get) interface{} {
-			return ds.driver
-		},
-	})
-
-	ds.controller.InitRestRoutes()
+	err = ds.selfRegister()
+	if err != nil {
+		ds.LoggingClient.Errorf("Failed to register service on Metadata: %v", err)
+		return false
+	}
 
 	err = provision.LoadProfiles(ds.config.Device.ProfilesDir, dic)
 	if err != nil {
-		ds.LoggingClient.Error(fmt.Sprintf("Failed to create the pre-defined device profiles: %v\n", err))
+		ds.LoggingClient.Errorf("Failed to create the pre-defined device profiles: %v", err)
 		return false
 	}
 
 	err = provision.LoadDevices(ds.config.DeviceList, dic)
 	if err != nil {
-		ds.LoggingClient.Error(fmt.Sprintf("Failed to create the pre-defined devices: %v\n", err))
+		ds.LoggingClient.Errorf("Failed to create the pre-defined devices: %v", err)
 		return false
 	}
 
 	ds.manager.StartAutoEvents()
-	http.TimeoutHandler(nil, time.Millisecond*time.Duration(ds.config.Service.Timeout), "Request timed out")
 
 	return true
 }

--- a/pkg/service/main.go
+++ b/pkg/service/main.go
@@ -47,6 +47,15 @@ func Main(serviceName string, serviceVersion string, proto interface{}, ctx cont
 		container.ConfigurationName: func(get di.Get) interface{} {
 			return ds.config
 		},
+		container.DeviceServiceName: func(get di.Get) interface{} {
+			return ds.deviceService
+		},
+		container.ProtocolDriverName: func(get di.Get) interface{} {
+			return ds.driver
+		},
+		container.ProtocolDiscoveryName: func(get di.Get) interface{} {
+			return ds.discovery
+		},
 	})
 
 	httpServer := handlers.NewHttpServer(router, true)

--- a/pkg/service/service.go
+++ b/pkg/service/service.go
@@ -121,20 +121,22 @@ func (s *DeviceService) DeviceDiscovery() bool {
 }
 
 // AddRoute allows leveraging the existing internal web server to add routes specific to Device Service.
-func (s *DeviceService) AddRoute(route string, handler func(http.ResponseWriter, *http.Request), methods ...string) error {
+func (s *DeviceService) AddRoute(route string, handler func(http.ResponseWriter, *http.Request), methods ...string) errors.EdgeX {
 	return s.controller.AddRoute(route, handler, methods...)
 }
 
 // Stop shuts down the Service
 func (s *DeviceService) Stop(force bool) {
 	if s.initialized {
-		_ = s.driver.Stop(false)
+		err := s.driver.Stop(force)
+		if err != nil {
+			s.LoggingClient.Error(err.Error())
+		}
 	}
-	s.manager.StopAutoEvents()
 }
 
 // selfRegister register device service itself onto metadata.
-func (s *DeviceService) selfRegister() error {
+func (s *DeviceService) selfRegister() errors.EdgeX {
 	newDeviceService := models.DeviceService{
 		Name:        s.ServiceName,
 		Labels:      s.config.Service.Labels,

